dts/bindings/misc/zephyr,flash-disk.yaml

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: Block storage device on flash partition
+
+compatible: "zephyr,flash-disk"
+
+include: base.yaml
+
+properties:
+    partition:
+      type: phandle
+      required: true
+      description: |
+        Backing storage flash map partition.
+
+    disk-name:
+      type: string
+      required: true
+      description: |
+        Disk name.
+
+    sector-size:
+      type: int
+      required: false
+      default: 512
+      description: |
+        Emulated block device sector size in bytes.
+
+    buffer-size:
```

Comment:
I think `min-buffer-size` is even more confusing that `buffer-size`. This parameter is not quite software configuration, but it is really related to underlying hardware. The problem is that this value cannot be (without adding significant hardware knowledge to devicetree parsing code) derived at compile time, so the user should provide it in the binding.

There is no actual effect on the system other that, "it works when it is large enough, and it does not work at all when it is too small". If the buffer is larger than necessary, flashdisk does not take any advantage of the extra space (i.e. it is just wasted).

Reply:
@de-nordic, I also think that min-buffer-size is confusing. When the "optimisation" of using a shared buffer would not be done it would just be a buffer-size and with the "optimization" it is checked that the size is correct.

Regarding the software/hardware configuration: this PR creates an emulation of a disk, a disk is a hardware construct, the emulation should therefore also be treated as a hardware construct (just like the flash emulator and the eeprom emulator). This also has the advantage that multiple disks can be created easily without any "software" coupling (except from the buffer, which is again a "optimisation" that could not have been done at all).

Reply:
The size should be checked at runtime regardless if there is shared buffer or not. The consequences of too small buffer lead to either data loss (i.e. some page data could not be preserved) or (as would be the case currently) overwriting memory after the buffer (what exactly depends on linker but doing so is not acceptable in any way).

Specifying the buffer on per-instance basis makes it possible to switch between shared and non-shared buffer later on without modifying devicetree entries, because the shared buffer is merely flashdisk implementation detail.